### PR TITLE
Add OpesSearch client to Elastic Search

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43504

## Description
Added support to not ElasticSearch server distributions

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 1]
```diff
@@ -21,6 +21,14 @@ configuration:
   name: proxy
   required: false
   type: 8
+- defaultvalue: Elasticsearch
```
Comment (reutshal) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:24]: Please add additional info here explaining who will need this and when

**Review by reutshal — CHANGES_REQUESTED** [review]
Please add unit-tests to run also with the open-search client (you can use parametrize).
* Please also add another configuration in content-test-conf so we will run the test playbook with the open-search client to make sure everything works as expected.

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py [hunk 2]
```diff
@@ -666,6 +673,10 @@ def main():
         elif demisto.command() == 'get-mapping-fields':
             get_mapping_fields_command()
     except Exception as e:
+        if 'The client noticed that the server is not a supported distribution of Elasticsearch' in str(e):
+            return_error('Failed executing {}. Seem that the client does not support the server\'s distribution, '
+                         'Please try using the Open Search client in the instance configuration.'
+                         '\nError message: {}'.format(demisto.command(), str(e)), error=e)
```
Comment (amshamah419) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.py:679]:
```suggestion
            return_error('Failed executing {}. Seems that the client does not support the server\'s distribution, '
                         'Please try using the Open Search client in the instance configuration.'
                         '\nError message: {}'.format(demisto.command(), str(e)), error=e)
```

File: Packs/Elasticsearch/ReleaseNotes/1_2_0.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Elasticsearch v2
+- Added the OpenSearch client
```
Comment (amshamah419) [inline Packs/Elasticsearch/ReleaseNotes/1_2_0.md:3]:
```suggestion
- Added support for the OpenSearch client.
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml (hunk 1 above)
Comment (amshamah419) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:24]:
```suggestion
- defaultvalue: Elasticsearch
  additionalInfo: In some hosted ElasticSearch environments (AWS), the standard ElasticSearch client is not supported. If your environment is not supported, please select OpenSearch as the parameter.
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 4]
```diff
@@ -21,6 +21,15 @@ configuration:
...
+  additionalinfo: In some hosted ElasticSearch environments (AWS), the standard ElasticSearch client is not supported. If your environment is not supported, please select OpenSearch as the parameter.
```
Comment (reutshal) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:25]:
```suggestion
  additionalinfo: In some hosted ElasticSearch environments (AWS), the standard ElasticSearch client is not supported. If your environment is not supported, please try using the OpenSearch client type.
```

Are we sure that it happens in AWS env?

Comment (reutshal) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:25]: Please also add the default value of 'Elasticsearch'.

File: Packs/Elasticsearch/ReleaseNotes/1_2_0.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Elasticsearch v2
+- Added the OpenSearch client to provide support for some hosted environments (AWS) which are not supported by the standard ElasticSearch client.
```
Comment (reutshal) [inline Packs/Elasticsearch/ReleaseNotes/1_2_0.md:3]:
```suggestion
- Added the OpenSearch client to provide support for some hosted environments (like AWS) which are not supported by the standard ElasticSearch client.
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py [hunk 6]
```diff
@@ -587,28 +631,34 @@ def test_format_to_iso():
...
     elasticsearch_builder(None)
     assert es_mock.call_args[1].get('http_auth') is None
     assert es_mock.call_args[1].get('api_key') is None
 
 
-def test_elasticsearch_parse_subtree():
+@pytest.mark.parametrize('params', MOCK_PARAMS)
+def test_elasticsearch_parse_subtree(params, mocker):
+    mocker.patch.object(demisto, 'params', return_value=params)
+    importlib.reload(Elasticsearch_v2)  # To reset the Elasticsearch client with the OpenSearch library
```
Comment (reutshal) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py:661]: Is this still checking both clients?

**Review by reutshal — CHANGES_REQUESTED** [review]
Nice, please see my comments

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py (hunk 6 above)
Comment (MosheEichler) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py:661]: Yes

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 7]
```diff
@@ -21,6 +21,15 @@ configuration:
   name: proxy
   required: false
   type: 8
+- defaultvalue: Elasticsearch
+  additionalinfo: In some hosted ElasticSearch environments, the standard ElasticSearch client is not supported. If you encounter any related client problems, please consider using the OpenSearch client type.
```
Comment (reutshal) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:25]:
```suggestion
  additionalinfo: In some hosted ElasticSearch environments, the standard ElasticSearch client is not supported. If you encounter any related client issues, please consider using the OpenSearch client type.
```

File: Packs/Elasticsearch/ReleaseNotes/1_2_0.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Elasticsearch v2
+- Added the OpenSearch client to provide support for some hosted environments (like AWS) which are not supported by the standard ElasticSearch client.
```
Comment (reutshal) [inline Packs/Elasticsearch/ReleaseNotes/1_2_0.md:3]:
```suggestion
- Added the OpenSearch client to provide support for some hosted environments which are not supported by the standard ElasticSearch client.
```

**Review by reutshal — CHANGES_REQUESTED** [review]
Please see final comments

**Review by reutshal — APPROVED** [review]

**Pull request merged**